### PR TITLE
Update to be compatible with Faraday

### PR DESCRIPTION
**Pull request by LukeIGS**
Related to https://github.com/savonrb/savon/pull/998

Changes some calls to faraday equivalents and updates the tests accordingly

### CODE REVIEW

File: lib/wasabi/resolver.rb [hunk 1]
```diff
@@ -41,10 +41,10 @@ def resolve
     private
 
     def load_from_remote
-      request.url = document
-      response = HTTPI.get(request, adapter)
+      request.adapter *adapter if adapter
+      response = request.get(document)
 
-      raise HTTPError.new("Error: #{response.code} for url #{request.url}", response) if response.error?
+      raise HTTPError.new("Error: #{response.status} for url #{document}", response) unless response.success?
```
Comment (pcai) [inline lib/wasabi/resolver.rb:47]: this uses the `Struct.new` syntax [introduced in ruby 3.2](https://bugs.ruby-lang.org/issues/16806), so it needs to be refactored to use the old style for compatibility with older rubies

**Review by pcai — CHANGES_REQUESTED** [review]
did some poking around and found some documentation on the order of parameters for Faraday::Env (which inherits from Options which extends Struct):

https://github.com/lostisland/faraday/blob/main/lib/faraday/options/env.rb#L57

Please make the changes for ruby support < 3.2 and I should be able to cut this as a new release

File: spec/support/responses.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Responses
+  class << self
+    def mock_faraday(code, headers, body)
+      env = Faraday::Env.new(status: code, response_headers: headers, response_body: body)
```
Comment (pcai) [inline spec/support/responses.rb:4]: this uses the `Struct.new` syntax [introduced in ruby 3.2](https://bugs.ruby-lang.org/issues/16806), so it needs to be refactored to use the old style for compatibility with older rubies

Comment (LukeIGS) [reply spec/support/responses.rb:4]: yeah, i was poking around, looked like it was because of how i was initializing for the unit test.

**Comment by LukeIGS** [comment]
Looks better.

**Comment by LukeIGS** [comment]
looks like the coveralls sever is utilizing tlsv1 which is making openssl very unhappy in the builds when coveralls tries to submit its results.

**Review by pcai — APPROVED** [review]

**Pull request merged**